chain/stmgr/events.go

```diff
@@ -0,0 +1,90 @@
+package stmgr
+
+import (
+	"context"
+	"github.com/ipfs/go-cid"
+
+	"github.com/filecoin-project/go-address"
+	"github.com/filecoin-project/lotus/api"
+	"github.com/filecoin-project/lotus/chain/events"
+	"github.com/filecoin-project/lotus/chain/store"
+	"github.com/filecoin-project/lotus/chain/types"
+	"github.com/filecoin-project/specs-actors/actors/abi"
+)
+
+type stateMgr interface {
+	GetActor(address.Address, *types.TipSet) (*types.Actor, error)
+	GetReceipt(context.Context, cid.Cid, *types.TipSet) (*types.MessageReceipt, error)
+}
+
+func NewEvents(ctx context.Context, chain *store.ChainStore, sm stateMgr) *events.Events {
+	return events.NewEvents(ctx, &eventsAdapter{
+		chain: chain,
+		sm: sm,
+	})
+}
+
+type eventsAdapter struct {
```

Comment:
I don't really like this, it duplicates a bunch of api impl, I'd try rather hard to avoid having to have this